modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AbstractValidator.java

```diff
@@ -113,6 +116,26 @@ private static void validateSCIMObjectForRequiredSubAttributes(AbstractAttribute
                             }
                         }
                     }
+
+                    // Check for canonical attributes in groups.
+                    List<String> canonicalValues = subAttributeSchema.getCanonicalValues();
+                    if (scimObject instanceof Group && canonicalValues != null) {
```

Comment:
what happens if the list is empty? can we validate the empty list scenario?

Reply:
addressed